icu4j/main/classes/core/src/com/ibm/icu/impl/number/DecimalQuantity_AbstractBCD.java

```diff
@@ -74,6 +74,12 @@
      */
     protected int origDelta;
 
+    /**
+     * The value of the (suppressed) exponent after the number has been put into
+     * a notation with exponents (ex: compact, scientific).
+     */
+    protected int exponent = 0;
```

Comment:
Make sure you handle this everywhere that you need; for example:

- copyFrom
- setBcdToZero

Reply:
Done.